kunta-api-server/src/test/java/fi/otavanopisto/kuntaapi/test/server/integrations/mikkelinyt/MikkeliNytRestTestsIT.java

```diff
@@ -0,0 +1,350 @@
+package fi.otavanopisto.kuntaapi.test.server.integrations.mikkelinyt;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.core.Is.is;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+
+import fi.otavanopisto.kuntaapi.server.integrations.IdType;
+import fi.otavanopisto.kuntaapi.server.integrations.mikkelinyt.MikkeliNytConsts;
+import fi.otavanopisto.kuntaapi.server.rest.model.Attachment;
+import fi.otavanopisto.kuntaapi.test.AbstractIntegrationTest;
+import fi.otavanopisto.mikkelinyt.model.Event;
+import fi.otavanopisto.mikkelinyt.model.EventsResponse;
+
+public class MikkeliNytRestTestsIT extends AbstractIntegrationTest {
+  
+  private static final String ORGANIZATION_SETTING_APIKEY = "test-api-key";
+  private static final String ORGANIZATION_SETTING_LOCATION = "test";
+  private static final int ATTACHMENT_SIZE = 38244;
+  private static final String ATTACHMENT_TYPE = "image/jpeg";
+  private static final ZoneId TIMEZONE_ID = ZoneId.of("Europe/Helsinki");
+  
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(getWireMockPort());
+
+  @Test
+  public void testEvents() {
+    String kuntaApiOrganizationId = "ka-organization-id";
+    String baseUrl = getWireMockBasePath();
+    String imagesBasePath = "uploads/testevents";
+    String kuntaApiEventId = "ka-event-id";
+    String kuntaApiAttachmentId = "ka-attachment-id";
+
+    createSettings(kuntaApiOrganizationId, baseUrl, imagesBasePath);
+    
+    MikkeliNytMocker mocker = new MikkeliNytMocker();
+    mockEvent(mocker, kuntaApiOrganizationId, kuntaApiEventId, kuntaApiAttachmentId, baseUrl, imagesBasePath);
+    mocker.startMock();
+    try {
+      given() 
+        .baseUri(getApiBasePath())
+        .contentType("application/json")
+        .get("/organizations/{ORGANIZATIONID}/events", kuntaApiOrganizationId)
+        .then()
+        .assertThat()
+        .statusCode(200)
+        .body("id.size()", is(1))
+        .body("id[0]", is(kuntaApiEventId))
+        .body("originalUrl[0]", is(String.format("%s/event-url-address", baseUrl)))
+        .body("name[0]", is("Test Event"))
+        .body("description[0]", is("My test<br/>event"))
+        .body("start[0]", sameInstant(getInstant(2020, 5, 6, 17, 30, TIMEZONE_ID)))
+        .body("end[0]", sameInstant(getInstant(2020, 5, 6, 19, 00, TIMEZONE_ID)))
+        .body("city[0]", is(ORGANIZATION_SETTING_LOCATION))
+        .body("place[0]", is("Testing Ltd"))
+        .body("address[0]", is("Testroad 3"))
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Testroad 3" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)